### PR TITLE
Fix residual gradient in non-learnable LayerNorm backward

### LINKED ISSUE #446: [BUG]Non-learnable LayerNorm backward drops residual gradient
**Describe the bug**

The non-learnable LayerNorm backward path in `examples/hstu/ops/triton_ops/triton_layer_norm.py` does not accumulate the residual gradient passed through `dx_accumulate`.

The learnable branch handles this gradient, but the non-learnable branch calls `_layer_norm_bwd_dx()` and returns immediately.

**Steps/Code to reproduce bug**

Use the non-learnable LayerNorm backward path with:

```python
input_norm_weight = None
dx_accumulate = residual_grad
```

The relevant code path is:

```python
_layer_norm_bwd_dx[(N,)](...)
return dx, None, None
```

`dx_accumulate` is not used before returning.

The returned gradient can be compared against:

```python
expected_dx = layer_norm_dx + dx_accumulate
```

**Expected behavior**

When `dx_accumulate` is provided, the returned input gradient should be:

```python
dx = layer_norm_dx + dx_accumulate
```

This should be consistent with the learnable LayerNorm backward path.

**Environment details (please complete the following information):**

* Environment location: Not applicable; identified through source-code review
* Method of recsys-examples install: From source
* `print_env.sh` output: Not available because I do not have access to a CUDA-capable NVIDIA GPU environment

**Additional context**

The issue is in the non-learnable branch of `triton_weighted_layer_norm_bwd()`.

A minimal fix would be:

```python
if dx_accumulate is not None:
    dx.add_(dx_accumulate)
```

before returning `dx`.

### PR DESCRIPTION
**Pull request by rocketbastard**
## Description

Fix residual gradient accumulation in the non-learnable LayerNorm backward path.

- Fuse `dx_accumulate` into the `_layer_norm_bwd_dx` Triton kernel.
- Remove the separate PyTorch `dx.add_()` operation.
- Add focused numerical unit tests for the LayerNorm operator.

Supersedes #447 because the original fork was deleted.
Fixes #446

## Testing

The added unit tests passed successfully in the local GPU environment.

## Checklist

- [x] I am familiar with the [Contributing Guidelines](https://github.com/NVIDIA/recsys-examples/blob/main/CONTRIBUTING.md).
- [x] New or existing tests cover these changes.
- [x] The documentation is up to date with these changes.

### CODE REVIEW

**Comment by JacoCheung** [comment]
Hi @rocketbastard could you move the test code to [hstu-specific test file](https://github.com/rocketbastard/recsys-examples/blob/130a6e896eadf01adfea74fdd8dbc1a3ccb8582a/examples/hstu/test/test_ln_mul_dropout.py) ?

**Comment by JacoCheung** [comment]
:x: **Pipeline #63652527** -- failed

| Job | Status | Log |
|-----|--------|-----|
| pre_check | :x: failed | [view](https://gitlab-master.nvidia.com/Devtech-Compute/distributed-recommender/-/jobs/404841146) |
| train_build_x86 | :white_check_mark: success | [view](https://gitlab-master.nvidia.com/Devtech-Compute/distributed-recommender/-/jobs/404841147) |
| train_build_arm64 | :white_check_mark: success | [view](https://gitlab-master.nvidia.com/Devtech-Compute/distributed-recommender/-/jobs/404841148) |
| prepare-jet-b200-smoke | :white_check_mark: success | [view](https://gitlab-master.nvidia.com/Devtech-Compute/distributed-recommender/-/jobs/404841149) |
| prepare-jet-b200-inference | :white_check_mark: success | [view](https://gitlab-master.nvidia.com/Devtech-Compute/distributed-recommender/-/jobs/404841151) |
| prepare-jet-cw-dfw-e2e-benchmark | :white_check_mark: success | [view](https://gitlab-master.nvidia.com/Devtech-Compute/distributed-recommender/-/jobs/404841153) |
| build_whl | :white_check_mark: success | [view](https://gitlab-master.nvidia.com/Devtech-Compute/distributed-recommender/-/jobs/404841155) |
| dynamicemb_test_fwd_bwd_8gpus | :white_check_mark: success | [view](https://gitlab-master.nvidia.com/Devtech-Compute/distributed-recommender/-/jobs/404841156) |
| dynamicemb_test_load_dump_8gpus | :white_check_mark: success | [view](https://gitlab-master.nvidia.com/Devtech-Compute/distributed-recommender/-/jobs/404841157) |
| unit_test_1gpu_a100 | :x: failed | [view](https://gitlab-master.nvidia.com/Devtech-Compute/distributed-recommender/-/jobs/404841159) |
| unit_test_1gpu_h100 | :x: failed | [view](https://gitlab-master.nvidia.com/Devtech-Compute/distributed-recommender/-/jobs/404841160) |
| unit_test_4gpu | :x: failed | [view](https://gitlab-master.nvidia.com/Devtech-Compute/distributed-recommender/-/jobs/404841161) |
| unit_test_tp_4gpu | :x: failed | [view](https://gitlab-master.nvidia.com/Devtech-Compute/distributed-recommender/-/jobs/404841162) |
| L20_unit_test_1gpu | :white_check_mark: success | [view](https://gitlab-master.nvidia.com/Devtech-Compute/distributed-recommender/-/jobs/404841163) |
| inference_unit_test_1gpu | :white_check_mark: success | [view](https://gitlab-master.nvidia.com/Devtech-Compute/distributed-recommender/-/jobs/404841164) |
| inference_test_1gpu | :white_check_mark: success | [view](https://gitlab-master.nvidia.com/Devtech-Compute/distributed-recommender/-/jobs/404841165) |

**Result: 11/16 jobs passed**

[View full pipeline](https://gitlab-master.nvidia.com/Devtech-Compute/distributed-recommender/-/pipelines/63652527)

**Review by shijieliu — APPROVED** [review]

**Review by shijieliu — APPROVED** [review]
